### ISSUE TITLE
"peer ID" and "node ID" should be consistent between restarts

### ISSUE THREAD

**Issue by feross** [feross/webtorrent#353]
The DHT needs to be persisted, and that means our DHT node ID shouldn't change between different runs of the webtorrent program.

Should we persist the main client's peer ID too?

Are there privacy implications to re-using the peer ID? What does Transmission do?

**Comment by Wingman4l7** [comment]
Definitely persist at least the peer ID *prefix*, or you won't be able to have webtorrent be a whitelisted client on private torrent trackers!

**Comment by feross** [comment]
Yep, we have a consistent prefix! It's `WW-`.

**Comment by Wingman4l7** [comment]
I thought so!  I might suggest that you append the version number of webtorrent to the prefix, so that a specific stable version of the client can be whitelisted.

**Comment by feross** [comment]
Yep, we already do that. You can see the code here: https://github.com/feross/webtorrent/blob/5f3e73f9937228fac0b9974de7db5b4fbb16b1bf/index.js#L61

**Comment by Wingman4l7** [comment]
... I think I knew that.  >.<  Well, great!  :D

**Comment by gillesdemey** [comment]
Transmission persists at least a part of the peerid, like `-TR1330-` and the rest of the 20 bytes are filled in at random, just like WebTorrent does it right now.

https://trac.transmissionbt.com/wiki/PeerId

They generate a new peerID every six hours though, see https://trac.transmissionbt.com/changeset/13933

**Comment by feross** [comment]
@gillesdemey 

That part of the Transmission ID (`-TR1330-`) identifies the client and the version. For example, WebTorrent uses `-WW-0057`.

Interesting that they generate a new ID every six hours. Perhaps that's to prevent long-term user tracking?

In any case, it looks like there's probably no benefit (and possibly privacy harm) to preserving the peer ID over the long term. Good to know.
